### ISSUE TITLE
Commit 6e5554a breaks linking in many projects

### ISSUE THREAD

**Issue by Cryptiiiic** [tihmstar/libgeneral#9]
example from futurerestore
![image](https://user-images.githubusercontent.com/27748705/148863228-ad3552b4-9a83-4f58-9d44-96b5b23a222d.png)

```
        try {
            ticketIdentity = img4tool::getBuildIdentityForIm4m({im4m.first,im4m.second}, buildmanifest);
        } catch (tihmstar::exception &e) {
            //
        }
```
6e5554a

**Comment by Cryptiiiic** [comment]
It seems to only not link for m1 arm64? Weird how x86_64 works. Clang shenanigans I guess?

**Comment by Cryptiiiic** [comment]
wow over a year later finally fixed lol
fixed in c68553a
